src/Middleware/WebSockets/src/WebSocketMiddleware.cs

```diff
@@ -165,6 +155,54 @@ public async Task<WebSocket> AcceptAsync(WebSocketAcceptContext acceptContext)
 
                 return WebSocket.CreateFromStream(opaqueTransport, isServer: true, subProtocol: subProtocol, keepAliveInterval: keepAliveInterval);
             }
+
+            public static bool CheckSupportedWebSocketRequest(string method, IHeaderDictionary requestHeaders)
+            {
+                if (!string.Equals("GET", method, StringComparison.OrdinalIgnoreCase))
+                {
+                    return false;
+                }
+
+                foreach (var pair in requestHeaders)
+                {
+                    if(!requestHeaders[HeaderNames.Connection].Contains(HeaderNames.Upgrade))
```

Comment:
I believe you need to keep `GetCommaSeparatedValues` for now, otherwise you can match "Upgrade2" header value.

Reply:
Hi @BrennanConroy, Sorry I totally forget the code here and also some context. Can you please provide the example here that I need to change and use `GetCommaSeparatedValues`. Thanks

Reply:
You need something like this for each header
```
for (var value in requestHeaders.GetCommaSeparatedValues(headerName))
{
  if (string.Equals(value, headerName, StringComparison.OrdinalIgnoreCase))
  {
  }
}
```

Reply:
Hi @BrennanConroy, Can you please pick up this from here? Because my system isn't working great, seems like a hard drive failure! Need to buy a new one ultimately and it may take some time. So I don't want to hang this PR left here as it's important! I think you can directly commit to my branch as well, as you people will be having permission. 

Sorry again! 😔🙏

Reply:
No worries, I've got it. Thanks for helping out.

Reply:
Thanks @Tratcher for kind help! 🙌🙌